Automattic/jetpack
# Replace copied social icons with social logos package

We have Social Icons copied at many places and it has been discussed many times before during PR reviews that we should use the [`social-logos`](https://github.com/Automattic/social-logos) package for the icons.

## Proposed changes:

* Install `"social-logos"` as dependency in `js-packages/components`
* Replace copied Social icons with `SocialLogo` component from `"social-logos"`
* Clean up `js-packages/components/components/icons` README
* Update social icons consumer components

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:

* Checkout this PR and run `pnpm install`
* Run `jetpack build plugins/jetpack`
* Create a post
* Click Jetpack icon on the top of the sidebar
* Under "Social Previews", confirm that icons look the same as before
* Click on "Preview"
* Confirm that icons look the same as before
* Under "Share this post", confirm that icons are the same as before
* Create a new post and hit publish or for an existing post, run this in browser console
    - `wp.data.dispatch('core/edit-post').togglePublishSidebar();`
* On the post publish panel, confirm that icons are the same as before
* Look for any other places where the social icons are used to confirm everything is OK.


<img width="288" alt="Screenshot 2023-10-16 at 12 16 53 PM" src="https://github.com/Automattic/jetpack/assets/18226415/191dc1d3-438e-4e53-804c-d925017d9c14">
<img width="1276" alt="Screenshot 2023-10-16 at 12 17 11 PM" src="https://github.com/Automattic/jetpack/assets/18226415/fae3297d-96aa-40a5-8cc0-ab63994d555e">
